File: server/test-goose.js
const OpenAI = require("openai");
const axios = require("axios");

// Load environment variables
require('dotenv').config();

console.log("Testing GooseAI integration...");
console.log("API Key:", process.env.KB_Key ? "Key is set" : "Key is missing");

// Test using direct axios call to GooseAI
async function testGooseAI() {
  try {
    console.log("Sending request to GooseAI using axios...");
    
    const response = await axios.post(
      'https://api.goose.ai/v1/engines/gpt-j-6b/completions',
      {
        prompt: "Roses are red",
        max_tokens: 25,
        temperature: 0.7
      },
      {
        headers: {
          'Authorization': `Bearer ${process.env.KB_Key}`,
          'Content-Type': 'application/json'
        }
      }
    );
    
    console.log("Response received:");
    console.log(response.data.choices[0].text);
  } catch (error) {
    console.error("Error occurred:");
    console.error(error.message);
    
    if (error.response) {
      console.error("Response data:", error.response.data);
      console.error("Response status:", error.response.status);
    }
  }
}

testGooseAI();

File: server/test-goose.py
import os
import openai
import dotenv

# Load environment variables
dotenv.load_dotenv()

# Get API key from environment
api_key = os.getenv("KB_Key")
print(f"API Key available: {'Yes' if api_key else 'No'}")

# Configure OpenAI with GooseAI
openai.api_key = api_key
openai.api_base = "https://api.goose.ai/v1"

try:
    # List Engines (Models)
    print("Listing available engines...")
    engines = openai.Engine.list()
    
    # Print all engines IDs
    print("Available engines:")
    for engine in engines.data:
        print(f"- {engine.id}")

    # Create a completion
    print("\nGenerating completion...")
    prompt = "Once upon a time there was a Goose. "
    print(f"Prompt: {prompt}")
    
    completion = openai.Completion.create(
        engine="gpt-j-6b",
        prompt=prompt,
        max_tokens=160,
        stream=False)  # Set to False for testing

    # Print the completion
    print("\nCompletion result:")
    print(completion.choices[0].text)

except Exception as e:
    print(f"\nError: {e}")
